app/components/ui/molecules/PeoplePicker.js

```diff
@@ -0,0 +1,159 @@
+import styled from 'styled-components'
+import { Button } from '@pubsweet/ui'
+import { th } from '@pubsweet/ui-toolkit'
+import { Box, Flex } from 'grid-styled'
+
+import SelectedItem from '../atoms/SelectedItem'
+import PersonPod from '../atoms/PersonPod'
+
+const SelectedItems = ({ selection, onCloseClick }) => (
+  <Flex>
+    {selection.map(person => (
+      <Box key={person.id} mr={1}>
+        <SelectedItem
+          label={person.name}
+          onCloseClick={() => onCloseClick(person)}
+        />
+      </Box>
+    ))}
+  </Flex>
+)
+
+const SuccessMessage = styled.div`
+  color: ${th('colorSuccess')};
+`
+
+const SelectionHint = ({ selection, minSelection, maxSelection }) => {
+  const selectionLength = selection.length
+  if (selectionLength < minSelection) {
+    const numRequired = minSelection - selectionLength
+    return (
+      <div>
+        {numRequired} more suggestion{numRequired === 1 ? '' : 's'} required
+      </div>
+    )
+  }
+  if (selectionLength >= maxSelection) {
+    return <SuccessMessage>Maximum {maxSelection} choices</SuccessMessage>
+  }
+  return null
+}
+
+const PeoplePickerBody = ({
+  isSelected,
+  maxSelection,
+  minSelection,
+  onCancel,
+  onSubmit,
+  people,
+  selection,
+  toggleSelection,
+}) => (
+  <React.Fragment>
+    <Flex justifyContent="space-between">
+      <SelectedItems onCloseClick={toggleSelection} selection={selection} />
+      <SelectionHint
+        maxSelection={maxSelection}
+        minSelection={minSelection}
+        selection={selection}
+      />
+    </Flex>
+    <Flex flexWrap="wrap" mx={-2}>
+      {people.map(person => (
+        <Box key={person.id} p={2} width={1 / 2}>
+          <PersonPod.ChosenPersonPod
+            iconState={isSelected(person) ? 'selected' : 'add'}
+            institution={person.institution}
+            isIconClickable={
+              isSelected(person) || selection.length < maxSelection
+            }
+            name={person.name}
+            onIconClick={() => toggleSelection(person)}
+          />
+        </Box>
+      ))}
+    </Flex>
+  </React.Fragment>
+)
+
+const PeoplePickerButtons = ({ isValid, onCancel, onSubmit }) => (
+  <Flex>
+    <Box mr={3}>
+      <Button onClick={onCancel}>Cancel</Button>
+    </Box>
+    <Box>
+      <Button disabled={!isValid} onClick={onSubmit} primary>
+        Add
+      </Button>
+    </Box>
+  </Flex>
+)
+
+class PeoplePicker extends React.Component {
+  state = { selection: [] }
+
+  toggleSelection(person) {
+    if (this.isSelected(person)) {
+      this.setState({
+        selection: this.state.selection.filter(p => p.id !== person.id),
+      })
+    } else {
+      this.setState({ selection: this.state.selection.concat(person) })
+    }
+  }
+
+  handleSubmit() {
+    if (this.isValid()) {
+      this.props.onSubmit(this.state.selection)
+    }
+  }
+
+  isSelected(person) {
+    return this.state.selection.some(p => p.id === person.id)
+  }
+
+  isValid() {
```

Comment:
What do you think of making this name more explicit? Like `hasReachedMaxLimit`?

Reply:
No I think this is valid (boom, boom). This is testing whether the picker selection is in a valid state – like a form. Currently that involves checking max and min but in future that could change.

Reply:
Then perhaps the name should change in the future too? :wink:

But seriously, if we're using a name that makes it sound like you can put whatever you want in there, perhaps this is an indication that we should allow people to add their own function to `isValid`?

Rather than this being a stateful component, perhaps we should make `isValid`, `isSelected`, `handleSubmit`, etc all just inputs to a functional component?